### ISSUE TITLE
The "Undefined offset: 1" error during every command when ImageMagic extension is found in PHP

### ISSUE THREAD

**Issue by aik099** [composer/composer#8055]
My `composer.json`:

```json
{
	"name": "project-name",
	"require": {
		"smalot/pdfparser": "dev-intechnic"
	},
	"require-dev": {
		"aik099/coding-standard": "dev-in-portal"
	},
	"repositories": [
		{
			"type": "vcs",
			"url": "https://github.com/intechnic/pdfparser"
		}
	],
	"config": {
		"platform": {
			"php": "5.4"
		}
	}
}
```

Output of `composer diagnose`:

```
Deprecation warning: Your package name project-name is invalid, it should have a vendor name, a forward slash, and a package name. The vendor and package name can be words separated by -, . or _. The complete name should match "[a-z0-9]([_.-]?[a-z0-9]+)*/[a-z0-9]([_.-]?[a-z0-9]+)*". Make sure you fix this as Composer 2.0 will error.
Checking composer.json: FAIL
name : invalid value (project-name), must match [A-Za-z0-9][A-Za-z0-9_.-]*/[A-Za-z0-9][A-Za-z0-9_.-]*
No license specified, it is recommended to do so. For closed-source software you may use "proprietary" as license.
Checking platform settings: The OpenSSL library (0.9.8e-fips-rhel5) used by PHP does not support TLSv1.2 or TLSv1.1.
If possible you should upgrade OpenSSL to version 1.0.1 or above.

Checking git settings: OK
Checking http connectivity to packagist: OK
Checking https connectivity to packagist: OK
Checking github.com rate limit: FAIL
[Composer\Downloader\TransportException] The "https://api.github.com/rate_limit" file could not be downloaded: SSL operation failed with code 1. OpenSSL Error messages:
error:1407742E:SSL routines:SSL23_GET_SERVER_HELLO:tlsv1 alert protocol version
Failed to enable crypto
failed to open stream: operation failed
Checking disk free space: OK
Checking pubkeys: FAIL
Missing pubkey for tags verification
Missing pubkey for dev verification
Run composer self-update --update-keys to set them up
Checking composer version: OK
Composer version: 1.8.4

                       
  [ErrorException]     
  Undefined offset: 1  
                       

diagnose
```

When I run this command:

```
composer install
```

I get the following output:

```
Deprecation warning: Your package name project-name is invalid, it should have a vendor name, a forward slash, and a package name. The vendor and package name can be words separated by -, . or _. The complete name should match "[a-z0-9]([_.-]?[a-z0-9]+)*/[a-z0-9]([_.-]?[a-z0-9]+)*". Make sure you fix this as Composer 2.0 will error.
Loading composer repositories with package information

                       
  [ErrorException]     
  Undefined offset: 1  
                       

install [--prefer-source] [--prefer-dist] [--dry-run] [--dev] [--no-dev] [--no-custom-installers] [--no-autoloader] [--no-scripts] [--no-progress] [--no-suggest] [-v|vv|vvv|--verbose] [-o|--optimize-autoloader] [-a|--classmap-authoritative] [--apcu-autoloader] [--ignore-platform-reqs] [--] [<packages>]...
```

And I expected this to happen:

No error and `vendor` folder gets updated.

Debugging revealed, that error happens during `imagemagick` extension version parsing, that introduced in #7762. Here is what's being parsed:

```
ImageMagick 6.2.8 05/07/12 Q16 file:/usr/share/ImageMagick-6.2.8/doc/index.html
```

and `$matches` array is empty (version is `6.2.8` instead of expected `6-2-8` I guess).

**Comment by glensc** [comment]
what is that os? 

looks like heavily patched against upstream, with several differences:
1. order of version and Quantum swapped
2. missing `-patchlevel`
3. date in locale specific format.

**Comment by aik099** [comment]
> what is that os?

It's `CentOS 5` and PHP 5.4 with all extensions installed from Remi repo if I'm not mistaken.

> looks like heavily patched against upstream

I have no idea what that means. What is patched (Composer or OS or ImageMagick extension).

**Comment by glensc** [comment]
Looks like imagick 6.x version string is different. here's sample from system where `ext-imagick` is built with 6.x:

```
imagick

imagick module => enabled
imagick module version => 3.1.2
imagick classes => Imagick, ImagickDraw, ImagickPixel, ImagickPixelIterator
ImageMagick version => ImageMagick 6.2.9 08/24/06 Q16 http://www.imagemagick.org
ImageMagick copyright => Copyright (C) 1999-2006 ImageMagick Studio LLC
ImageMagick release date => 08/24/06
ImageMagick number of supported formats:  => 134
ImageMagick supported formats => A, ART, AVI, AVS, B, BMP, BMP2, BMP3, C, CACHE, CAPTION, CIN, CIP, CLIP, CMYK, CMYKA, CUR, CUT, DCM, DCX, DFONT, DNG, DPX, EPI, EPS, EPSF, EPSI, EPT, EPT2, EPT3, FAX, FITS, FRACTAL, G, G3, GIF, GIF87, GRADIENT, GRAY, HISTOGRAM, ICB, ICO, ICON, JNG, JPEG, JPG, K, M, M2V, MAP, MAT, MATTE, MNG, MONO, MPC, MPEG, MPG, MTV, NULL, O, OTB, OTF, P7, PAL, PALM, PAM, PATTERN, PBM, PCD, PCDS, PCL, PCT, PCX, PDB, PFA, PFB, PGM, PICON, PICT, PIX, PJPEG, PLASMA, PNG, PNG24, PNG32, PNG8, PNM, PPM, PREVIEW, PS, PSD, PTIF, PWP, R, RAS, RGB, RGBA, RGBO, RLA, RLE, SCR, SCT, SFW, SGI, STEGANO, SUN, TEXT, TGA, TIFF, TILE, TIM, TTC, TTF, TXT, UIL, UYVY, VDA, VICAR, VID, VIFF, VST, WBMP, WPG, X, XBM, XC, XCF, XPM, XV, XWD, Y, YCbCr, YCbCrA, YUV

Directive => Local Value => Master Value
imagick.locale_fix => 0 => 0
imagick.progress_monitor => 0 => 0
```

I might work on this in the evening, but pasting output, if somebody else wants to update patterns to match.

**Comment by glensc** [comment]
created sooner: https://github.com/composer/composer/pull/8056

**Comment by Seldaek** [comment]
Fixed by #8056
